### PR TITLE
Allow users to get a list of trackables

### PR DESCRIPTION
**Pull request by JakubJankowski**
Currently it's impossible to know which trackables are tracked by a publisher.
This PR adds a delegate method that returns a list of trackables that are currently added to a publisher whenever this list changes (similarly to how it works on Android).

Resolves: [336](https://github.com/ably/ably-asset-tracking-swift/issues/336)

### CODE REVIEW

**Review by KacperKluka — APPROVED** [review]
LGTM in terms of functionality :+1:

File: Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift [hunk 1]
```diff
@@ -851,6 +856,13 @@ extension DefaultPublisher {
             self.delegate?.publisher(sender: self, didUpdateResolution: event.resolution)
         }
     }
+    
+    private func notifyTrackablesChanged(trackables: Set<Trackable>) {
+        performOnMainThread { [weak self] in
```
Comment (ikbalkaya) [inline Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift:861]: Just a question, wouldn't this be a bit more suited to put on code where main thread is performing ? Such as in UI code

**Review by ikbalkaya — COMMENTED** [review]
Just a question

File: Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift (hunk 1 above)
Comment (JakubJankowski) [reply Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift:861]: Not sure I understand the question, but generally - `performOnMainThread` runs the code on the main, UI thread:
```swift
DispatchQueue.main.async(execute: operation)
```
We're running similar delegate methods on this same thread already as well, like:
notifyDelegateEnhancedLocationChanged,
notifyDelegateConnectionStateChanged,
notifyDelegateResolutionUpdate

etc.

Comment (ikbalkaya) [reply Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift:861]: I meant that I would expect the delegates  to perform this on main thread but not the `DefaultPublisher` itself. Unless operations are confined on main thread. But I see that this pattern has been used elsewhere in this class so I think it's ok for now.

**Review by ikbalkaya — APPROVED** [review]

**Pull request merged**